RestaurantReservation.Db/Repositories/CustomersRepository.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Microsoft.EntityFrameworkCore;
+using RestaurantReservation.Db.Models;
+
+namespace RestaurantReservation.Db.Repositories
+{
+    public class CustomersRepository
+    {
+        private RestaurantReservationDbContext _context;
+
+        public CustomersRepository(RestaurantReservationDbContext context)
+        {
+            _context = context;
+            _context.Database.EnsureCreatedAsync().Wait();
+        }
+
+        public DbSet<CustomerDTO> DbSet => _context.Customers;
+
+        /// <returns>The ID of the created object.</returns>
+        public async Task<int> CreateAsync(CustomerDTO newCustomer)
+        {
+            var customer = await _context.Customers.AddAsync(newCustomer);
+            await _context.SaveChangesAsync();
+            return customer.Entity.Id;
+        }
+
+        /// <exception cref="KeyNotFoundException"></exception>
+        public async Task<CustomerDTO> GetAsync(int customerId)
+        {
+            var customer = await _context.Customers
+                .SingleOrDefaultAsync(customer => customer.Id == customerId);
+
+            if (customer != null)
+            {
+                return customer;
+            }
+            else
+            {
+                throw new KeyNotFoundException(
+                    $"Customer with ID = {customerId} does not exist.");
+            }
+        }
+
+        public async Task<List<CustomerDTO>> GetAllAsync()
+        {
+            return await _context.Customers.ToListAsync();
+        }
+
+        public async Task UpdateAsync(CustomerDTO updatedCustomer)
+        {
+            _context.Customers.Update(updatedCustomer);
```

Comment:
let's validate here also. 
u can use the same validator

Reply:
done